### DISCUSSION TITLE
Downloa and stream simultaneously but stream the downlaoded file.

### DISCUSSION THREAD

**Discussion by tathastu871** [streamlink/streamlink#5158] [Q&A]
### Description

i am streaming video url or m3u8 in mpv or vlc android. Via termux.
streamlink -Q --hls-segment-stream-data --ringbuffer-size 100M --stream-segment-threads 10 --config=/dev/null --player-continuous-http --player-http --player=bash -a '-c '\''am start -n is.xyz.mpv/.MPVActivity -a android.intent.action.VIEW -d '\''{filename}'\'' ; read -n 1'\''' "$url" best;

eg http://127.0.0.1:8000/file.m3u8
when played it shows file.m3u8 as title in vlc or mpv
i wmat to stRt downloading file.m3u8 and save as customtitle.ts or mp4 something easiky playable and stream that to vlc or mpv

i want streamlink to start downlaoding and use that downladed file to stream **simultaneously** to vlc or mpv **using -r or -R optionsuch that the title of stream being played on vlc or mpv can be customised**

**Comment by bastimeyer** [comment]
> --player-passthrough hls,http

https://streamlink.github.io/cli.html#cmdoption-player-passthrough
> **A comma-delimited list of stream types to pass to the player as a URL to let it handle the transport of the stream instead.**

----

You are completely bypassing Streamlink by using the passthrough parameter. This is the equivalent of
`"$player" "$(streamlink --stream-url "$URL" "$quality")"`
meaning that you're making Streamlink launch the player with the resolved URL as first parameter instead of letting Streamlink use its own stream implementation and letting the player read from it. Other stream-related options like `-o`/`-r` etc. won't have any effect.

**Reply by tathastu871** [reply]
removed player passthrough
but  no effect it sets tite to url 127.0.0.1:somenumber

any help with customizing title.
on pc --title may work but on android it does  not. 
its like
streamlink http://127.0.0.1:8000/file.m3u8 -o custom_name.mp4
playert custom_name.mp4

but want in single run. start downloading and use that being downlaoded file as input to player so that player set title to input file which can be customised with -o -r or -R options

help pleaE

**Reply by bastimeyer** [reply]
> any help with customizing title.

https://mpv.io/manual/stable/#options-force-media-title

**Reply by tathastu871** [reply]
mpv on android does not support custom title unless root. thats why i asked on here

**Reply by tathastu871** [reply]
streamlink -Q --hls-segment-stream-data --ringbuffer-size 100M --stream-segment-threads 10 --config=/dev/null --player-continuous-http --player-http --player=bash -a '-c '''am start -n is.xyz.mpv/.MPVActivity -a android.intent.action.VIEW -d '''{filename}''' ; read -n 1'''' "http://127.0.0.1:8000/file.m3u8" best;

sets the title to file.m3u8. 
can it be customized. beacsue streamlink must be downlaofng streM at temp locTion befotr streaming to player. so that temp location file can be renamed to custom one


What i am saying is i can do
streamlink url best -o custom.mp4 & #background it
player custom.mp4

but then player only player some part of vidro that has been downlaoded at time of opening eg first 5 min only it will not updTe with realtime downloaded stream


does streamlink start downloding file and save in temportay file. and that temporory file name can be cutomisef and can be strea.med realtime to player while downlaoding. player will set title to tempororay filename which is customised one i want

**Reply by gravyboat** [reply]
@tathastu871 I'm giving you this as a final warning. Almost no question you have asked has been related to Streamlnk and I'm getting tired of seeing them. This is for Streamlink related discussion and questions, not questions for how you can use us as free technical support for your custom setup. I'm going to lock this and if I see another off-topic post from you I'm going to block you from being able to post. This is a small team with limited time. We don't have the resources to help people with whatever custom setup they may want to run, it's outside the scope of the project. Read the documentation and investigate your unique needs on your own.
